hatchery/nextflow.go

```diff
@@ -1075,11 +1082,33 @@ func setupSubnet(subnetName string, cidr string, vpcid string, ec2Svc *ec2.EC2)
 		return exsubnet.Subnets[0].SubnetId, nil
 	}
 
+	// Fetch all availability zones
+	// this is being limited to a region by the ec2svc that gets passed in.
+	describeZonesOutput, err := ec2Svc.DescribeAvailabilityZones(&ec2.DescribeAvailabilityZonesInput{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to describe availability zones: %v", err)
+	}
+
+	// Avoid us-east-1e as it is technically neglected, and does not have modern instance types.
+	// Find a suitable availability zone that is not us-east-1e
```

Comment:
can we mention the exact problem we ran into in this comment, for future reference - the exact instance type we're using that isn't compatible, as of june 2024

Reply:
It's in the PR comments, there is no "official" documentation of this problem though.

Reply:
iirc the problem is with `instance-type=g4dn.xlarge`. It's not mentioned in the comment

Reply:
Added another comment.

Reply:
I guess we shouldn't hard-code the actual name of the AZ `us-east-1e`: because `1e` in one AWS account does not correspond to `1e` in another AWS account.

I'd prefer if we do some check that for instance type to be available in AZ, rather than hard-coding AZ per AWS account.

Reply:
I think you might be confusing the zonename with zone id, zone id (example `use1-az2`) will point to different AZ's.  Zone names are consistent across all accounts. 

https://docs.aws.amazon.com/wo rkspaces/latest/adminguide/azs-workspaces.html

I do like the idea of making sure the instance exists, instead of hardcoding though.

Reply:
I am referring to this:
>For example, the Availability Zone us-east-1a for your AWS account might not be the same location as us-east-1a for another AWS account.

And [this](https://docs.aws.amazon.com/ram/latest/userguide/working-with-az-ids.html):
>AWS maps the physical Availability Zones randomly to the Availability Zone names for each AWS account.

I am fine with current fix, but I think it should be clearer on why it happens, and the fact that in different AWS account it would be different, when and if Nextflow would be used there.

Reply:
We need to convert this aws cli command to our sdk code: 

```
aws ec2 describe-instance-type-offerings \                                                                                                                                                                                      
    --location-type availability-zone \
    --filters Name=location,Values=us-east-1a Name=instance-type,Values=g4dn.xlarge
```

Reply:
@jawadqur -- Is this being built as a part of this PR? or is the PR ready for review again?

Reply:
I've not had bw to work more on this PR this week. 

I guess we need to make a decision if this is "good enough" or implement the approach mentioned above. 

@paulineribeyre / @m0nhawk - What are your thoughts?

Reply:
I am fine with this 👍  — my only point is keeping track of that, so in the future when this is used in another AWS account and we have the same issue in let's say `us-east-1c` we know why and fix it.

Reply:
I created a ticket: https://ctds-planx.atlassian.net/browse/MIDRC-768